hadoop-crypto/src/main/java/com/palantir/hadoop/cipher/FsCipherInputStream.java

```diff
@@ -103,15 +103,25 @@ public boolean seekToNewSource(long targetPos) throws IOException {
     // NOTE(jellis): may want to implement the other read methods depending on performance
     @Override
     public int read() throws IOException {
-        int ret = decryptedStream.read();
-        if (ret != -1) {
+        int bytesRead = decryptedStream.read();
+        if (bytesRead != -1) {
             decryptedStreamPos++;
         }
-        return ret;
+        return bytesRead;
+    }
+
+    @Override
+    public int read(byte[] buffer, int offset, int length) throws IOException {
+        int bytesRead = decryptedStream.read(buffer, offset, length);
+        if (bytesRead != -1) {
+            decryptedStreamPos += bytesRead;
+        }
+        return bytesRead;
     }
 
     @Override
     public void close() throws IOException {
         delegate.close();
+        decryptedStream.close();
```

Comment:
We could split this out to a separate PR, but it seems incorrect that we're not closing the CipherInputStream which finalizes the Cipher.